middleware/http/opa/middleware.go

```diff
@@ -43,6 +48,32 @@ var (
 	errOpaInvalidResultType = errors.New("got an invalid type back from repo policy. Only a boolean or map is valid")
 )
 
+func (s *Status) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return nil
+	}
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
```

Comment:
I'm guessing that this has to be `float64` because that's what `json.Unmarshal` converts string numbers to and not `int` ... thank you so much for adding the test that catches these cases!

Because this is a float value now though, we should trap illegal inputs here instead of silently truncating, so add a check for that:
``` go
if value != math.Trunc(value) {
	return fmt.Errorf("invalid float value %f parse to status(int)", value)
}
```

Reply:
Done.